src/api.ts

```diff
@@ -0,0 +1,387 @@
+import * as _ from "lodash";
+import * as querystring from "querystring";
+import * as requestModule from "request";
+import * as url from "url";
+import * as FormData from "form-data";
+
+import { Constants } from "./emulator/constants";
+import { FirebaseError } from "./error";
+import * as logger from "./logger";
+import * as responseToError from "./responseToError";
+import * as scopes from "./scopes";
+import * as utils from "./utils";
+
+export enum Method {
+  GET = "GET",
+  PUT = "PUT",
+  POST = "POST",
+  DELETE = "DELETE",
+  PATCH = "PATCH",
+}
+
+type Origin = string;
```

Comment:
Why not just a `string`? I think we may be overdoing it on the types here...

Reply:
Yeah, this may be a bit overkill (seeing the comments below too)